framework/src/main/java/io/github/kgress/scaffold/AutomationWait.java

```diff
@@ -233,12 +236,26 @@ public Boolean waitForElementToNotHaveClass(BaseWebElement element, String class
      * @return                      as {@link Boolean}
      */
     public Boolean waitUntilPageIsLoaded(Long setTempTimeout) {
+        return waitUntilPageIsLoaded(setTempTimeout, 0);
+    }
+
+
+    private Boolean waitUntilPageIsLoaded(Long setTempTimeout, int retryCount) {
         var domReadyStateScript = "return document.readyState";
-        return waitForCustomCondition(page -> getWebDriverWrapper()
-                .getJavascriptExecutor()
-                .executeScript(domReadyStateScript)
-                .equals("complete"),
-                setTempTimeout);
+        try {
+            return waitForCustomCondition(page -> getWebDriverWrapper()
+                            .getJavascriptExecutor()
+                            .executeScript(domReadyStateScript)
+                            .equals("complete"),
+                    setTempTimeout);
+        } catch (WebDriverException e) {
+            if (e.getMessage().contains("target frame detached") && retryCount < 5) {
+                AutomationUtils.sleep(200);
+                return waitUntilPageIsLoaded(setTempTimeout, ++retryCount);
+            } else {
+                throw(e);
+            }
+        }
```

Comment:
Could you add javadoc here on why we are catching this specific message, and other general details?